stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -0,0 +1,230 @@
+"""Database logic."""
+import logging
+from typing import List, Type, Union
+
+import attr
+import elasticsearch
+from elasticsearch_dsl import Q, Search
+
+from stac_fastapi.elasticsearch import serializers
+from stac_fastapi.elasticsearch.config import ElasticsearchSettings
+from stac_fastapi.types.errors import NotFoundError
+from stac_fastapi.types.stac import Collection, Collections, Item, ItemCollection
+
+logger = logging.getLogger(__name__)
+
+NumType = Union[float, int]
+
+ITEMS_INDEX = "stac_items"
+COLLECTIONS_INDEX = "stac_collections"
+
+
+def mk_item_id(item_id: str, collection_id: str):
+    """Make the Elasticsearch document _id value from the Item id and collection."""
+    return f"{item_id}|{collection_id}"
+
+
+@attr.s
+class CoreDatabaseLogic:
+    """Core database logic."""
+
+    settings = ElasticsearchSettings()
+    client = settings.create_client
+    item_serializer: Type[serializers.Serializer] = attr.ib(
+        default=serializers.ItemSerializer
+    )
+    collection_serializer: Type[serializers.Serializer] = attr.ib(
+        default=serializers.CollectionSerializer
+    )
+
+    @staticmethod
+    def bbox2poly(b0, b1, b2, b3):
+        """Transform bbox to polygon."""
+        poly = [[[b0, b1], [b2, b1], [b2, b3], [b0, b3], [b0, b1]]]
+        return poly
+
+    def get_all_collections(self, base_url: str) -> Collections:
+        """Database logic to retrieve a list of all collections."""
+        try:
+            collections = self.client.search(
+                index=COLLECTIONS_INDEX, query={"match_all": {}}
+            )
+        except elasticsearch.exceptions.NotFoundError:
+            raise NotFoundError("No collections exist")
```

Comment:
Should this return an empty `Collections` instead of raising?

Reply:
True, that would make sense. I would have to ask @philvarner maybe.

Reply:
yes, an empty Collections rather than 404

Reply:
maybe file an issue and put that in a separate PR instead of this one -- just focus on the reorganization here.

Reply:
Ok for sure